deepchem/models/lightning/dc_lightning_module.py

```diff
@@ -24,35 +71,68 @@ def __init__(self, dc_model):
-
-        self.pt_model = self.dc_model.model
-        # This will not work for ModularTorchModel as it is directly uses `loss_func` to compute loss.
-        self.loss = self.dc_model._loss_fn
+        self.pt_model = dc_model.model
+        self.loss: LossFn = dc_model._loss_fn
+        self.optimizer: Optimizer = dc_model.optimizer
+        self._prediction_outputs: Optional[
+            List[int]] = dc_model._prediction_outputs
+        self._loss_outputs: Optional[List[int]] = dc_model._loss_outputs
+        self._variance_outputs: Optional[List[int]] = dc_model._variance_outputs
+        self._other_outputs: Optional[List[int]] = dc_model._other_outputs
+        self.uncertainty: bool = getattr(dc_model, 'uncertainty', False)
+        self.learning_rate: Union[float,
+                                  LearningRateSchedule] = dc_model.learning_rate
+        self.transformers: List[Transformer] = []
+        self.other_output_types: Optional[OneOrMany[str]] = None
 
     def configure_optimizers(self):
-        return self.dc_model.optimizer._create_pytorch_optimizer(
-            self.pt_model.parameters(),)
+        """Configure optimizers and learning rate schedulers.
+
+        Returns
+        -------
+        Union[torch.optim.Optimizer, Tuple[List[torch.optim.Optimizer], List[torch.optim.lr_scheduler.LRScheduler]]]
+            PyTorch optimizer or tuple containing lists of optimizers and schedulers.
+        """
+        self.dc_model._built = True
+        pytorch_optimizer = self.optimizer._create_pytorch_optimizer(
+            self.pt_model.parameters())
 
-    def training_step(self, batch, batch_idx):
+        if isinstance(self.optimizer.learning_rate, LearningRateSchedule):
+            lr_schedule = self.optimizer.learning_rate._create_pytorch_schedule(
+                pytorch_optimizer)
+            return [pytorch_optimizer], [lr_schedule]
+
+        return pytorch_optimizer
+
+    def training_step(self, batch: Union[Tuple[Any, Any, Any], Any],
+                      batch_idx: int):
         """Perform a training step.
 
         Parameters
         ----------
-        batch: A tensor, tuple or list.
-        batch_idx: Integer displaying index of this batch
-        optimizer_idx: When using multiple optimizers, this argument will also be present.
+        batch: Union[Tuple[Any, Any, Any], Any]
+            A tensor, tuple or list containing inputs, labels, and weights.
+        batch_idx: int
+            Integer displaying index of this batch
 
         Returns
         -------
-        loss_outputs: outputs of losses.
+        torch.Tensor
+            The computed loss tensor for this training step.
         """
-        batch = batch.batch_list
-        inputs, labels, weights = self.dc_model._prepare_batch(batch)
+        if hasattr(batch, 'batch_list'):
+            # If batch is a DCLightningDatasetBatch, extract the batch list.
+            batch = batch.batch_list  # type: ignore
```

Comment:
is this the type of batch_list -> Tuple[List, List, List]  since this this the input type for prepare batch. Please update accordingly and maybe do not assign batch_list to the variable `batch` intead make it as `batch_list` maybe